src/data/option/basic.lean

```diff
@@ -415,6 +415,13 @@ by cases o; refl
   get_or_else (o.map f) (f x) = f (get_or_else o x) :=
 by cases o; refl
 
+/-- `option α` is a `subsingleton` if and only if `α` is empty. -/
+lemma subsingleton_iff_is_empty : subsingleton (option α) ↔ is_empty α :=
+⟨λ h, ⟨λ x, some_ne_none x $ @subsingleton.elim _ h _ _⟩,
+  λ h, ⟨λ x y, option.cases_on x (option.cases_on y rfl (λ x, h.elim x)) (λ x, h.elim x)⟩⟩
+
+instance [is_empty α] : subsingleton (option α) := subsingleton_iff_is_empty.2 ‹_›
```

Comment:
Can you make this a stronger `unique` instance?

Reply:
Currently neither of `data.option.basic` and `logic.unique` import the other. Which import should I add?